{
    "metadata": {
        "@context": "https://openvex.dev/ns/v0.2.0",
        "@id": "https://github.com/canonical/ubuntu-security-notices/blob/main/vex/usn/USN-5138-1",
        "author": "Canonical Ltd.",
        "timestamp": "2024-07-31T20:28:25.705146",
        "version": 1
    },
    "statements": [
        {
            "vulnerability": {
                "@id": "https://ubuntu.com/security/notices/USN-5138-1",
                "name": "USN-5138-1",
                "description": "The py.path.svnwc component of py (aka python-py) through v1.9.0 contains  a regular expression with an ambiguous subpattern that is susceptible to  catastrophic backtracing. This could be used by attackers to cause a compute-time  denial of service attack by supplying malicious input to the blame functionality.",
                "aliases": [
                    "CVE-2020-29651"
                ]
            },
            "timestamp": "2021-11-10 23:09:54.489651",
            "products": [
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.5.2-1ubuntu0.1?arch=src?distro=ubuntu/bionic"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.5.2-1ubuntu0.1?arch=amd64"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.5.2-1ubuntu0.1?arch=arm64"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.5.2-1ubuntu0.1?arch=armhf"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.5.2-1ubuntu0.1?arch=i386"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.5.2-1ubuntu0.1?arch=ppc64el"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.5.2-1ubuntu0.1?arch=s390x"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.5.2-1ubuntu0.1?arch=amd64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.5.2-1ubuntu0.1?arch=arm64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.5.2-1ubuntu0.1?arch=armhf"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.5.2-1ubuntu0.1?arch=i386"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.5.2-1ubuntu0.1?arch=ppc64el"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.5.2-1ubuntu0.1?arch=s390x"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.5.2-1ubuntu0.1?arch=amd64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.5.2-1ubuntu0.1?arch=arm64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.5.2-1ubuntu0.1?arch=armhf"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.5.2-1ubuntu0.1?arch=i386"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.5.2-1ubuntu0.1?arch=ppc64el"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.5.2-1ubuntu0.1?arch=s390x"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.8.1-1ubuntu0.1?arch=src?distro=ubuntu/focal"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.8.1-1ubuntu0.1?arch=amd64"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.8.1-1ubuntu0.1?arch=arm64"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.8.1-1ubuntu0.1?arch=armhf"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.8.1-1ubuntu0.1?arch=i386"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.8.1-1ubuntu0.1?arch=ppc64el"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.8.1-1ubuntu0.1?arch=riscv64"
                },
                {
                    "@id": "pkg:deb/ubuntu/pypy-py@1.8.1-1ubuntu0.1?arch=s390x"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.8.1-1ubuntu0.1?arch=amd64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.8.1-1ubuntu0.1?arch=arm64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.8.1-1ubuntu0.1?arch=armhf"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.8.1-1ubuntu0.1?arch=i386"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.8.1-1ubuntu0.1?arch=ppc64el"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.8.1-1ubuntu0.1?arch=riscv64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python-py@1.8.1-1ubuntu0.1?arch=s390x"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.8.1-1ubuntu0.1?arch=amd64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.8.1-1ubuntu0.1?arch=arm64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.8.1-1ubuntu0.1?arch=armhf"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.8.1-1ubuntu0.1?arch=i386"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.8.1-1ubuntu0.1?arch=ppc64el"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.8.1-1ubuntu0.1?arch=riscv64"
                },
                {
                    "@id": "pkg:deb/ubuntu/python3-py@1.8.1-1ubuntu0.1?arch=s390x"
                }
            ],
            "status": "fixed",
            "status_notes": "python-py could be made to crash if provided with specially crafted input."
        }
    ]
}